specification/apimanagement/control-plane/Microsoft.ApiManagement/preview/2017-03-01/apimapis.json

```diff
@@ -0,0 +1,1878 @@
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ApiExportResult": {
+      "properties": {
+        "link": {
+          "type": "string",
+          "description": "Link to the Storage Blob containing the result of the export operation. The Blob Uri is only valid for 5 minutes."
+        }
+      },
+      "description": "API Export result Blob Uri."
+    },
+    "ApiCollection": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ApiContract"
+          }
+        },
+        "count": {
+          "type": "integer",
+          "description": "Total number of entities"
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Next page link if any."
+        }
+      },
+      "description": "Paged Api list representation."
+    },
+    "ApiCreateOrUpdateParameter": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Identifier of the entity"
+        },
+        "contentValue": {
+          "type": "string",
+          "description": "Content value when Importing an API."
+        },
+        "contentFormat": {
+          "type": "string",
+          "description": "Format of the Content in which the API is getting imported.",
+          "enum": [
+            "wadl-xml",
+            "wadl-link-json",
+            "swagger-json",
+            "swagger-link-json",
+            "wsdl",
+            "wsdl-link"
+          ],
+          "x-ms-enum": {
+            "name": "ContentFormat",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "wadl-xml",
+                "description": "The contents are inline and Content type is a WADL document."
+              },
+              {
+                "value": "wadl-link-json",
+                "description": "The WADL document is hosted on a publicly accessible internet address."
+              },
+              {
+                "value": "swagger-json",
+                "description": "The contents are inline and Content Type if a OpenApi 2.0 Document."
+              },
+              {
+                "value": "swagger-link-json",
+                "description": "The Open Api 2.0 document is hosted on a publicly accessible internet address."
+              },
+              {
+                "value": "wsdl",
+                "description": "The contents are inline and the document is a WSDL/Soap document."
+              },
+              {
+                "value": "wsdl-link",
+                "description": "The WSDL document is hosted on a publicly accessible internet address."
+              }
+            ]
+          }
+        },
+        "wsdlSelector": {
```

Comment:
>wsdlSelector [](start = 9, length = 12)

For a better code generation, it is a good idea to not define stuff inline.

Reply:
sorry did not understand what you mean.

---
In reply to: [163341864](https://github.com/Azure/azure-rest-api-specs/pull/2269#discussion_r163341864) [](ancestors = 163341864)

Reply:
Instead of defining `wsdlSelector` and its properties here, it could have its own definition, and just be referenced here.

---
In reply to: [163346532](https://github.com/Azure/azure-rest-api-specs/pull/2269#discussion_r163346532) [](ancestors = 163346532,163341864)